src/catalog/src/durable/impls/persist/state_update.rs

```diff
@@ -542,124 +541,81 @@ impl RustType<proto::StateUpdateKind> for StateUpdateKind {
     }
 }
 
-/// Binary version of [`StateUpdateKind`] to allow reading/writing raw binary from/to persist.
-#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Arbitrary)]
-pub(crate) struct StateUpdateKindBinary(pub(crate) Vec<u8>);
-
-#[derive(Debug, Clone, Default)]
-pub struct StateUpdateKindSchema;
-
-impl From<StateUpdateKind> for StateUpdateKindBinary {
-    fn from(value: StateUpdateKind) -> Self {
-        Self(value.into_proto().encode_to_vec())
+impl StateUpdateKind {
+    pub fn to_json_row(&self) -> SourceData {
+        let kind = self.into_proto();
+        let kind_raw = StateUpdateKindRaw::from_serde(&kind);
+        kind_raw.0
     }
-}
 
-impl TryFrom<StateUpdateKindBinary> for StateUpdateKind {
-    type Error = String;
-
-    fn try_from(value: StateUpdateKindBinary) -> Result<Self, Self::Error> {
-        let kind =
-            proto::StateUpdateKind::decode(value.0.as_slice()).map_err(|err| err.to_string())?;
-        kind.into_rust().map_err(|err| err.to_string())
+    pub fn from_json_row(row: SourceData) -> Self {
+        let kind = StateUpdateKindRaw(row).to_serde();
+        StateUpdateKind::from_proto(kind).expect("WIP")
     }
 }
-impl Codec for StateUpdateKindBinary {
-    type Schema = StateUpdateKindSchema;
-
-    fn codec_name() -> String {
-        "StateUpdateProto".to_string()
-    }
 
-    fn encode<B>(&self, buf: &mut B)
-    where
-        B: bytes::BufMut,
-    {
-        buf.put(self.0.as_slice());
-    }
+/// Version of [`StateUpdateKind`] to allow reading/writing raw json from/to persist.
+///
+/// WIP might be nice for this to be Row instead of SourceData, see how that feels
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Arbitrary)]
+pub(crate) struct StateUpdateKindRaw(pub(crate) SourceData);
 
-    fn decode<'a>(buf: &'a [u8]) -> Result<Self, String> {
-        Ok(Self(buf.to_vec()))
+impl From<StateUpdateKind> for StateUpdateKindRaw {
+    fn from(value: StateUpdateKind) -> Self {
+        Self(value.to_json_row())
     }
 }
 
-impl mz_persist_types::columnar::Schema<StateUpdateKindBinary> for StateUpdateKindSchema {
-    type Encoder<'a> = SimpleEncoder<'a, StateUpdateKindBinary, Vec<u8>>;
-
-    type Decoder<'a> = SimpleDecoder<'a, StateUpdateKindBinary, Vec<u8>>;
+impl TryFrom<StateUpdateKindRaw> for StateUpdateKind {
+    type Error = String;
 
-    fn columns(&self) -> DynStructCfg {
-        SimpleSchema::<StateUpdateKindBinary, Vec<u8>>::columns(&())
+    fn try_from(value: StateUpdateKindRaw) -> Result<Self, Self::Error> {
+        Ok(StateUpdateKind::from_json_row(value.0))
     }
+}
```

Comment:
If these are all infallible, then we can simplify the trait to use `From` instead of `TryFrom`.

Reply:
Ended up leaving these as TryFrom (but keeping the error as before)